Backend/DataAccessLayer/Repositories/UserManagementRepository.cs

```diff
@@ -67,5 +67,17 @@ public bool ExistingUser(DatabaseContext _db, string email)
             }
             return false;
         }
+
+        public bool IsManagerOf(DatabaseContext _db, User user, User subject)
+        {
+            if (subject == null || subject.ManagerId == null) return false;
+
+            if (subject.ManagerId == user.Id) return true;
+
+            // Check if we're a manager of the subjects manager
+            var managerOfSubject = GetUser(_db, (Guid) subject.ManagerId);
+            if (managerOfSubject == null) return false;
+            return IsManagerOf(_db, user, managerOfSubject);

```

Comment:
The purpose of this last bit throws me off a bit, if we send in originally two User objects to find out if user object is a manager of subjectuser object, wouldn't it be confusing if this function recursively called itself and returned true for the user we didn't originally specify?

Reply:
This is the recursive call that walks up the tree. This is the whole point of the tree. This is the part that supports multiple levels of ownership.

Say the MGMT chain looks like this:

Me
  |
Sam
  |
Fred

Without that line, I could only check if I have management over Sam. With that line, I could check if I have management over Fred (via Sam).

This is what Vong described.